staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go

```diff
@@ -297,34 +298,48 @@ func (o *DrainCmdOptions) RunDrain() error {
 	}
 
 	drainedNodes := sets.NewString()
-	var fatal error
+	var deleteOrEvictErrors []error
+	var deleteOrEvictResultMutex sync.Mutex
```

Comment:
If you use a channel instead, you won't need `deleteOrEvictErrors`, `deleteOrEvictResultMutex`, or `deleteOrEvictWaitGroup`. Did you consider that way?

Reply:
Ah yeah I should've thought to use a channel, that would be nicer. I can revisit the code with that approach

Reply:
I'm working through the details of this now and I'm not certain using channels is the best approach. If the goal is to preserve the existing behavior of printing the success or failure of draining each node, and then printing the list of nodes that still need to be drained at the end, and then to return drain errors then I think I need three channels. One for successfully drained nodes, one for unsuccessfully drained nodes and one for errors. I also still need the wait group so that I can block until all nodes have either succeeded or failed at draining.

Is there something I'm missing that could simplify the channel solution?

Reply:
Hmm, I don't know for sure, but I was originally thinking something like this:
```
var results = make(chan error)
for _, info := range o.nodeInfos {
	info := info
	go func() { results <- o.deleteOrEvictPodsSimple(info) }()
}

for range o.nodeInfos {
	err := <- results
	// TODO: Do everything here that was being done in the original for loop
}
```

But I guess you also need `info` coming out of the channel too, so maybe that wouldn't really make it simpler because you would need a struct or something to return.

Probably fine to use waitgroup in this case

Reply:
Gotcha yeah I think the output requirements makes the waitgroup a bit simpler thann the channel option. I can go the struct route though if you'd prefer it

Reply:
No it’s fine, I was just thinking it might be simpler for a second, but waitgroup is probably fine

Reply:
Either solution would be good, I'd personally lean towards solution proposed by Brian here. You'd need two channels one for errors and another for node names. But that's not something I'd block a PR on.